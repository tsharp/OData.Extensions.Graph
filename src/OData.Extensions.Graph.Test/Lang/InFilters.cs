
using HotChocolate.Execution;
using HotChocolate.Language;
using OData.Extensions.Graph.Lang;
using Snapshooter.Xunit;
using System.Diagnostics;
using System.Threading.Tasks;
using Xunit;

namespace OData.Extensions.Graph.Test.Lang
{
    public class InFilters
    {
        [Fact]
        public static void InString()
        {
            var query = "{user(where: { Id: { in: [ \"Milk\", \"Cheese\" ] } }) { items { Id } } }";

            Utf8GraphQLParser.Parse(query).ToString(true).MatchSnapshot();

            // Arrange
            var translator = new QueryTranslator(Common.GetEdmModel());

            // Act
            var filerByUserId = translator.Translate("/user?$select=Id&$filter=Id in ('Milk', 'Cheese')");

            // Assert
            filerByUserId.DocumentNode.ToString(true).MatchSnapshot();
        }
    }
}
