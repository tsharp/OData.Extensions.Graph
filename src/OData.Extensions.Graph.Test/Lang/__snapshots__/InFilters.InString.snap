{
  user(where: { Id: { in: [ "Milk", "Cheese" ] } }) {
    items {
      Id
    }
  }
}
